api/app/helpers/seeds_handler.rb

```diff
@@ -0,0 +1,231 @@
+module SeedsHandler
+    def request(action, path, params = nil)
+        '''
+        action: :get, :post, :put, :delete
+        path: no need to include "/api/v1" portion of the route
+        '''
+        begin
+            env = { method: action, params: params }
+            mock_session = Rack::MockSession.new(Rails.application)
+            session = Rack::Test::Session.new(mock_session)
+            session.request("/api/v1#{path}", env)
+            return JSON.parse(mock_session.last_response.body, symbolize_names: true)
+        rescue JSON::ParserError
+            error = "#{action.to_s.upcase} #{path} is an invalid path."
+            return {status: 'error', message: error, payload: {}}
+        end
+    end
+
+    def read_json(file, log)
+        seed_data_dir = Rails.root.join('db', 'seed')
+        file_dir = seed_data_dir+file
+        begin
+            data = File.read(file_dir)
+            return JSON.parse(data, symbolize_names: true), log
+        rescue Errno::ENOENT => e
+            log.push(e)
+            return [], log
+        rescue JSON::ParserError 
+            log.push("#{file_dir} is not a valid JSON file.")
+            return [], log
+        end
+    end
+
+    def insert_data(seed)
```

Comment:
I don't understand why you need all these functions...all your data should be seeded by making calls to the api with `request`. You should never need to peek inside any controllers...

Reply:
So, we're chaining request outcome to make more requests?